core/roboconf-dm-rest-services/src/main/java/net/roboconf/dm/rest/services/internal/resources/impl/ManagementResource.java

```diff
@@ -279,4 +292,102 @@ public Response shutdownApplication( String applicationName ) {
 
 		return result;
 	}
+
+
+	@Override
+	public Response setImage(final String name, final String qualifier, final InputStream image,
+							 final FormDataContentDisposition fileDetail) {
+
+		// Do set the image, and wrap exceptions in a HTTP response.
+		Response response;
+		try {
+			doSetImage(name, qualifier, image, fileDetail);
+			response = Response.noContent().build();
+		} catch (final NoSuchElementException | IllegalArgumentException e) {
```

Comment:
Nope I do not agree! 404 is when the resource does not exist. Here the resource is the `ManagementResource`, which always exists. Getting a 404 when the request params are wrong is very confusing, as it means the resource does not exists, instead of bad request params.

Would `name` and `qualifier` be `@PathParams`, 404 would have been appropriated, be in this particular case, it is not.